data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/cnec/CsaProfileCnecCreator.java

```diff
@@ -92,13 +97,26 @@ private void addFlowCnec(PropertyBag assessedElementPropertyBag) {
         String assessedSystemOperator = assessedElementPropertyBag.getId(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_OPERATOR);
         String assessedElementName = CsaProfileCracUtils.getUniqueName(assessedSystemOperator, nativeAssessedElementName);
 
-        FlowCnecAdder flowCnecAdder = crac.newFlowCnec();
-        flowCnecAdder
-            .withMonitored(false)
-            .withOptimized(true)
-            .withReliabilityMargin(0);
+        CnecAdder cnecAdder;
+        if (CsaProfileConstants.LimitType.CURRENT.equals(limitType)) {
+            cnecAdder = crac.newFlowCnec()
+                .withMonitored(false)
+                .withOptimized(true)
+                .withReliabilityMargin(0);
 
-        if (!this.addCurrentLimit(assessedElementId, flowCnecAdder, currentLimit, isCombinableWithContingency)) {
+            if (!this.addCurrentLimit(assessedElementId, (FlowCnecAdder) cnecAdder, isCombinableWithContingency)) {
+                return;
+            }
+        } else if (CsaProfileConstants.LimitType.VOLTAGE.equals(limitType)) {
+            cnecAdder = crac.newVoltageCnec()
+                .withMonitored(false)
```

Comment:
not really important,but to be coherent with CIM importer:
```suggestion
                .withMonitored(true)
```